src/project.ts

```diff
@@ -167,4 +167,84 @@ async function addEditorSettingsButton () {
 	}
 }
 
-export { addProgramInfo, keyboardShortcuts, addEditorSettingsButton, checkHiddenOrDeleted };
+function saveNewScratchpadToLocalStorage () {
+	let aceEditor;
+	let checkIfEditorIsReadyInterval = setInterval(function () {
+		aceEditor = document.getElementsByClassName("scratchpad-ace-editor")[0]?.env?.editor;
+
+		if (aceEditor !== undefined) {
+			// get the type of the program
+			const programType = window.location.href.split("/")[5];
```

Comment:
The `onProgramPage` has a parameter containing the current scratchpad's data.  If you pass the Program object into `saveNewScratchpadToLocalStorage`, you can read the scratchpad type via the `userAuthoredContentType` property.